upgrader.go

```diff
@@ -158,8 +158,13 @@ func (u *Upgrader) Upgrade() error {
 	}
 
 	if u.parent != nil {
+		// verify clean exit
 		select {
-		case <-u.parent.exited:
+		case err := <-u.parent.exited:
```

Comment:
In this case, only the first call to `Upgrade()` will return an error. All following calls will succeed.

I think that all calls to Upgrade() should return an error if the parent did something unexpected. This way we don't panic, but force the operator to take an action (probably fixing the code and then scheduling down time).

Reply:
The parent does no longer close extited on parent loss.